### PR TITLE
RCAL-513 Update uneven ramp fitting documentation

### PR DESCRIPTION
**Pull request by stscieisenhamer**
Resolves [RCAL-513](https://jira.stsci.edu/browse/RCAL-513)

Closes #

This PR addresses ...

**Checklist**
- [x] added entry in `CHANGES.rst` under the corresponding subsection
- ~updated relevant tests~
- [x] updated relevant documentation
- [ ] updated relevant milestone(s)
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below. [How to run regression tests on a PR](https://github.com/spacetelescope/romancal/wiki/Running-Regression-Tests-Against-PR-Branches)

### CODE REVIEW

**Comment by stscieisenhamer** [comment]
@schlafly Still work-in-progress. However, I wanted to get a review of the description section. I have added the information that is in the paper and should correspond to the `stcal.ramp_fitting.ols_cas22.fit_ramps` code.

However, there is a bit more math, going from the segments to the resultants, that occurs in `stcal.ramp_fitting.ols_cas22.fit_ramps_casertano` after the `fit_ramps` is called, basically combining the segments back to their resultants. I can "math this out", but was wondering if there is a source paper (or something) that is better to use to match terminology.

There are still technical issues, such as filling out the parameters for the step, etc. that need to be completed, but wanted you to get an early look at this.

**Comment by stscieisenhamer** [comment]
Rest of the formulae have been documented. This will definitely need re-review. Also update the parameter documentation.

**Comment by stscieisenhamer** [comment]
[Regression test](https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/437/) though it appears something is up with them, and this is documentation only anyways.

File: docs/roman/ramp_fitting/description.rst [hunk 1]
```diff
@@ -156,64 +118,80 @@ sufficient; they are given as:
...
+:math:`R_i` and the coefficients :math:`K_i`:
 
-Exposure-level computations:
-----------------------------
+.. math::
+   \hat F = \sum_{i} K_i R_i
+
+The calculation is skipped for pixels that have :math:`D = 0`. Note that the coefficient
+:math:`K_i` vanishes for each resultant that has a bad pixel, as a consequence of :math:`W_i`
+vanishing.
```
Comment (schlafly) [inline docs/roman/ramp_fitting/description.rst:154]: I don't really think of this in this way.  The W_i are what you specify above and never vanish.  But bad entries in the DQ array are used to split the ramps into segments, and within each segment, a separate fit is done, where all of the W_i are nominal.

File: docs/roman/ramp_fitting/description.rst [hunk 2]
```diff
@@ -156,64 +118,80 @@ sufficient; they are given as:
...
 
-where the sum is over all segments.
+The signal variance, :math:`V_S`, of the count rate in the signal-based component of the slope
+variance is computed as:
 
+.. math::
+   V_S = \sum_{i=0}^{N-1} {K_i^2 \tau_i} + \sum_{i<j} {2 K_i K_j \cdot \bar t_i}
 
-The variance of the slope due to Poisson noise is:
+Total variance, if desired, is a (biased) estimate of the total slope variance :math:`V` can
```
Comment (schlafly) [inline docs/roman/ramp_fitting/description.rst:167]: Let's leave off "biased."  There is a small bias due to the way the weights P are selected, which leads to a bias in the flux and therefore in the total variance.  But if we're going to talk about the biases people care much more about the O(0.1%) bias in the flux than in the O(0.1%) bias in the variance.

File: docs/roman/ramp_fitting/description.rst [hunk 3]
```diff
@@ -156,64 +118,80 @@ sufficient; they are given as:
...
 
 .. math::
-   var^P_{o} = \frac{1}{ \sum_{s} \frac{1}{ var^P_{s}}}
+   V = V_R + V_S \cdot \hat F
+
+Exposure-level computations:
+----------------------------
 
-The combined variance of the slope is the sum of the variances:
+The ramps for each resultant are reconstructed from it's segments, :math:`i`,
```
Comment (schlafly) [inline docs/roman/ramp_fitting/description.rst:176]:
```suggestion
The ramps for each resultant are reconstructed from its segments, :math:`i`,
```

**Review by schlafly — CHANGES_REQUESTED** [review]
This looks good.  I left a few minor comments.  The most important one is to check the denominator of the variances; I think some parentheses are misplaced.  Thanks!

File: docs/roman/ramp_fitting/description.rst [hunk 4]
```diff
@@ -156,64 +118,80 @@ sufficient; they are given as:
...
 
+The estimated slope, :math:`\hat F`, is computed as a sum over the resultants
+:math:`R_i` and the coefficients :math:`K_i`:
 
-Exposure-level computations:
-----------------------------
+.. math::
+   \hat F = \sum_{i} K_i R_i
+
+The calculation is skipped for pixels that have :math:`D = 0`. Note that the coefficient
```
[the remark this answers is not in the document]
Comment (stscieisenhamer) [reply docs/roman/ramp_fitting/description.rst]: removed this specific statement and added on about no fitting occurring when there is only one segment.

File: docs/roman/ramp_fitting/description.rst (hunk 2 above)
Comment (stscieisenhamer) [reply docs/roman/ramp_fitting/description.rst]: removed

File: docs/roman/ramp_fitting/description.rst (hunk 3 above)
Comment (stscieisenhamer) [reply docs/roman/ramp_fitting/description.rst]: fixed

File: docs/roman/ramp_fitting/description.rst [hunk 5]
```diff
@@ -156,64 +118,80 @@ sufficient; they are given as:
...
-The square root of the combined variance is stored in the ERR array of the primary output.
+   \hat F_{mean} &= \frac {\sum_i {w_i \hat F_i}} {\sum_i w_i}
 
-The overall slope depends on the slope and the combined variance of the slope of all
-segments, so is a sum over segments:
+The read noise is determined as follows:
 
 .. math::
-    slope_{o} = \frac{ \sum_{s}{ \frac{slope_{s}} {var^C_{s}}}} { \sum_{s}{ \frac{1} {var^C_{s}}}}
+   V_{R_{mean}} = \frac {\sum_i {w_i ^ 2 V_{R_i}}} {\sum_i {w_i ^ 2}}
```
[the remark this answers is not in the document]
Comment (stscieisenhamer) [reply docs/roman/ramp_fitting/description.rst]: over-aggressive squaring. fixed.

File: docs/roman/ramp_fitting/description.rst [hunk 6]
```diff
@@ -156,64 +118,78 @@ sufficient; they are given as:
...
 +-------------------+------------------------+----------+
 
-Segment-specific Computations:
-------------------------------
-The variance of the slope of a segment due to read noise is:
+Segment-Specific Computations
+-----------------------------
+
+The segment fitting implementation is based on Section 5 of Casertano et.
+al. 2022. If there is only one segment, no fitting is performed.
```
Comment (schlafly) [inline docs/roman/ramp_fitting/description.rst:125]: I think of a 'segment' as a block of consecutive resultants with no jumps in between.  One segment is fine; e.g., this is the case for a ramp with no cosmic rays.  A segment with one resultant isn't fine and gets skipped.

File: docs/roman/ramp_fitting/description.rst [hunk 7]
```diff
@@ -13,59 +15,34 @@ more detail below.
...
+The implementation is what is described in this document.
+
+Segments
+++++++++
+
+Segments are determined using the 3-D GROUPDQ array of the input data set, under
+the assumption that the jump step will have already flagged CR's. Segments are
+terminated where saturation flags are found. Pixels are processed simultaneously
+in blocks. The size of the block depends on the image size and the number of
+resultants.
```
Comment (schlafly) [inline docs/roman/ramp_fitting/description.rst:36]:
```suggestion
terminated where saturation flags are found.
```
We just process pixels one by one.

**Review by schlafly — COMMENTED** [review]
Sorry, a few more minor nits, but looks good!

File: docs/roman/ramp_fitting/description.rst (hunk 6 above)
Comment (stscieisenhamer) [reply docs/roman/ramp_fitting/description.rst:125]: Nope, I got the terminology wrong. Trying again...

**Review by schlafly — APPROVED** [review]
LGTM, thanks!

**Pull request merged**